src/mixins/actionText.js

```diff
@@ -59,6 +66,10 @@ export default {
 			 * @type {Event}
 			 */
 			this.$emit('click', event)
+
+			if (this.closeAfterClick) {
+				this.$parent.closeMenu()
```

Comment:
Will not work if the component is wrapped :thinking: 
I would use the internal Get parent function https://github.com/nextcloud/nextcloud-vue/blob/master/src/utils/GetParent.js

Reply:
What do you mean with wrapped? Could you give a short example so I understand and can check if getParent fixes the issue?

Reply:
I think I get it. Will adjust the PR.

Reply:
Yes! :)
If you create an api allowing external apps to register new actions into a menu (like in server in  the public sharing menu, or in the users management, in the user row end of line menu) external apps will provide their component: like ActionButton. But the app will want to have their own actions handlers!

So it will look like 
```vue
<CustomAction>
	<ActionButton @click="dosomething" />
</CustomAction>
```
And then 
```vue
<Actions>
	<component :is="CustomAction" />
</Actions>
```

Reply:
Fixed it.